core/src/main/java/org/infinispan/commands/write/PutMapCommand.java

```diff
@@ -251,8 +261,8 @@ public boolean ignoreCommandOnStatus(ComponentStatus status) {
    }
 
    @Override
-   public boolean readsExistingValues() {
-      return false;
+   public LoadType loadType() {
+      return hasFlag(Flag.IGNORE_RETURN_VALUES) ? LoadType.DONT_LOAD : LoadType.PRIMARY;
```

Comment:
Interceptors work on commands and context. Do you want a special flag in context that will force loading return values, and all the interceptors will have to check that, too? I don't like that idea at all.
Seems that Index.ALL could really require that on all owners, which is pretty annoying and not limited to PutMapCommand. And it should not work now either if persistence is involved.